core/lib/config/src/configs/wallets.rs

```diff
@@ -0,0 +1,71 @@
+use zksync_basic_types::{Address, H256};
+use zksync_crypto_primitives::PackedEthSignature;
+
+#[derive(Debug, Clone)]
+pub struct Wallet {
+    address: Address,
+    private_key: Option<H256>,
+}
+
+impl Wallet {
+    pub fn from_address(address: Address) -> Self {
+        Self {
+            address,
+            private_key: None,
+        }
+    }
+
+    pub fn from_private_key(private_key: H256, address: Option<Address>) -> anyhow::Result<Self> {
+        let calculated_address = PackedEthSignature::address_from_private_key(&private_key)?;
+        if let Some(address) = address {
+            if calculated_address != address {
+                anyhow::bail!("Malformed wallet, address doesn't correspond private_key")
+            }
+        }
+
+        Ok(Self {
+            address: calculated_address,
+            private_key: Some(private_key),
+        })
+    }
+
+    pub fn address(&self) -> Address {
+        self.address
+    }
+    pub fn private_key(&self) -> Option<H256> {
+        self.private_key
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct EthSender {
+    pub operator: Wallet,
+    pub blob_operator: Option<Wallet>,
+}
+
+#[derive(Debug, Clone)]
+pub struct StateKeeper {
+    pub fee_account: Wallet,
+}
+
+#[derive(Debug, Clone)]
+pub struct Wallets {
+    pub eth_sender: Option<EthSender>,
+    pub state_keeper: Option<StateKeeper>,
+}
```

Comment:
I'm not sure if I like the implications here.
First, the fee account private key is not needed by the core application. It's but an ability to express its presence sounds like a downside to me -- more ground to create problems out of nowhere.
Second, the semantics of "optional" private keys feel slightly off. It was done like this previously because it was a quickfix, but I'm not sure if we want to harden this kind of relationships. Two components may require a `Wallet` object: one may just need an address to read info, and one may be actually signing transactions, but you won't know from the constructor signature alone.
Third, configuration errors are not caught by this. If a component requires a private key to operate, but only address was provided, the config would successfully deserialized and the application would fail further in their lifecycle.
Finally, overall this kind of grouping seems natural on the first glance (these are accounts, accounts always have an address, but for some of them we don't know a private key), but I fail to see actual benefits. The PK and address may come from different sources in real life environment (public/secret configs), the information they expose have different levels of sensitivity (as stated above), some configurations can not be even expressed in theory (if an account is a smart wallet, we may have an address but it has no private key).

Probably I would much prefer keeping addresses and private keys separately, and ideally with clear indication of sensitivity (e.g. `EthSenderPrivateKey { key: H256 }` and `EthSender { /* other public configs */ }`.

Reply:
I also understand that in dev configs the PK and address would come together and it may be convenient to represent it this way from the dev environment POV, but we should design against the _operating_ environment.

Reply:
to add to this - we'll likely have the eth-sender as a separate service with separate deployments and configs. Please keep this in mind when designing the config system

Reply:
It's completely possible to use only part of the config

Reply:
Slightly changed the layout of protobuf